### ISSUE TITLE
Nyx - totalWeight can be wrong and this can cause wrong voting results

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2024-06-velocimeter-judging#394]
Nyx

Medium

# totalWeight can be wrong and this can cause wrong voting results

## Summary
If the user proceeds with the vote only once and does nothing afterward, the user's voting power does not decrease, and this affects the overall voting result.
## Vulnerability Detail
In this protocol, voting power linearly decreases over time.

```solidity
function _reset(uint _tokenId) internal {
        address[] storage _poolVote = poolVote[_tokenId];
        uint _poolVoteCnt = _poolVote.length;
        uint256 _totalWeight = 0;

        for (uint i = 0; i < _poolVoteCnt; i ++) {
            address _pool = _poolVote[i];
            uint256 _votes = votes[_tokenId][_pool];

            if (_votes != 0) {
                _updateFor(gauges[_pool]);
                weights[_pool] -= _votes;
                votes[_tokenId][_pool] -= _votes;
                if (_votes > 0) {
                    IBribe(external_bribes[gauges[_pool]])._withdraw(uint256(_votes), _tokenId);
                    _totalWeight += _votes;
                } else {
                    _totalWeight -= _votes;
                }
                emit Abstained(_tokenId, _votes);
            }
        }
        totalWeight -= uint256(_totalWeight);
        usedWeights[_tokenId] = 0;
        delete poolVote[_tokenId];
    }
```

However, if no action is taken for the tokenId after the first vote, the voting power of that token remains within totalWeight and weights because the _reset function will not be called. This directly affects the _ratio within the Voter.notifyRewardAmount function. It also impacts claimable in the Voter. updateFor function.

```solidity
function notifyRewardAmount(uint amount) external {
        require(msg.sender == minter,"not a minter");
        activeGaugeNumber = 0;
        currentEpochRewardAmount = amount;
        _safeTransferFrom(base, msg.sender, address(this), amount); // transfer the distro in
        uint256 _ratio = amount * 1e18 / totalWeight; // 1e18 adjustment is removed during claim
        
        if (_ratio > 0) {
            index += _ratio;
        }
        emit NotifyReward(msg.sender, base, amount);
    }
```

When the Voter.distribute function is called, newly minted tokens are distributed to each gauge according to the voting results.

Ultimately, the voting results differ from what was intended, affecting reward distribution.
## Impact
Voting results are manipulated, leading to wrong results.
## Code Snippet
https://github.com/sherlock-audit/2024-06-velocimeter/blob/main/v4-contracts/contracts/Voter.sol#L485-L495

https://github.com/sherlock-audit/2024-06-velocimeter/blob/main/v4-contracts/contracts/Voter.sol#L207-L232
## Tool used

Manual Review

## Recommendation

Duplicate of #171
